lib/fog/openstack.rb

```diff
@@ -433,7 +434,7 @@ def self.retrieve_tokens_v3(options, connection_options = {})
         request[:omit_default_port] = omit_default_port unless omit_default_port.nil?
 
         response = connection.request(request)
-        @@token_cache[{body: request_body, path: path}] = response, Time.now + 30 # 30-second TTL, enough for most requests
+        @@token_cache[{:body => request_body, :path => path}] = response, Time.now + 30 if cache_ttl > 0
```

Comment:
Maybe I missed something, but why not
`@@token_cache[{:body => request_body, :path => path}] = response, Time.now + cache_ttl if cache_ttl > 0`